lms/services/jwt.py

```diff
@@ -51,3 +57,44 @@ def encode_with_secret(cls, payload: dict, secret, lifetime):
         jwt_str = jwt.encode(payload, secret, algorithm="HS256")
 
         return jwt_str
+
+    @staticmethod
+    @lru_cache
+    def _get_jwk_client(jwk_url: str):
+        """
+        Get a PyJWKClient for the given key set URL.
+
+        PyJWKClient maintains a cache of keys it has seen we want to keep
+        the clients around with `lru_cache` in case we can reuse that internal cache
+        """
+        return jwt.PyJWKClient(jwk_url)
+
+    def decode_lti_token(self, id_token):
```

Comment:
This is a bigish method now. 

I could break some parts (eg `_verify_header`, _get_registration_for_token`) etc but I don't think is that big that that would benefit the readability.

Reply:
It's fairly big, but I think a little white space management might help the readability (examples below).

As a public method it should be above the one above. This could benefit from a docstring.